libs/blocks/vimeo/vimeo.js

```diff
@@ -5,15 +5,31 @@ import { createIntersectionObserver, createTag, getConfig, isInTextNode, loadLin
 class LiteVimeo extends HTMLElement {
   static preconnected = false;
 
+  static defaultTitle = 'Content from Vimeo';
+
   connectedCallback() {
     this.isMobile = navigator.userAgent.includes('Mobi');
     this.videoId = this.getAttribute('videoid');
+    this.title = this.getAttribute('title');
     this.setupThumbnail();
     this.setupPlayButton();
     this.addEventListener('pointerover', LiteVimeo.warmConnections, { once: true });
     this.addEventListener('click', this.addIframe);
   }
 
+  async fetchVideoTitle() {
+    if (this.title !== LiteVimeo.defaultTitle) return null;
+
+    try {
+      const response = await fetch(`https://vimeo.com/api/oembed.json?url=https://vimeo.com/${this.videoId}`);
+      const data = await response.json();
+      if (data.title && this.iframeEl) this.iframeEl.title = data.title;
+      return null;
+    } catch (error) {
+      return null;
```

Comment:
Nit: I think ESlint would accept simple `return` statements, as long as they're consistent throughout the function.